File: pset6/credit/credit.py
from cs50 import get_int
import math


def main():
    # prompt for input
    while True:
        card = get_int("Number: ")
        if card > 0:
            break

    # help vaiables
    counter, numbers, help1, befor_last, last = 0, 0, 0, 0, 0
    temp = card
    # to change once multuply and once not
    x = False
    # do throught each number
    while temp > 0:
        numbers += 1
        if temp < 100 and temp > 9:
            befor_last = temp % 10

        elif temp < 10 and temp >= 0:
            last = temp
        # divide to go to the second number
        help1 = temp % 10
        temp = math.trunc(temp / 10)
        if x:
            help1 *= 2
            if help1 > 9:
                help1 -= 9

            counter += help1

        else:
            counter += help1

        x = not x
    # if the card is real find which company
    if counter % 10 == 0:
        if numbers == 15 and last == 3 and (befor_last == 4 or befor_last == 7):
            print("AMEX")

        elif (numbers == 16 or numbers == 13) and last == 4:
            print("VISA")

        elif numbers == 16 and last == 5 and (befor_last == 1 or befor_last == 2 or befor_last == 3 or befor_last == 4 or befor_last == 5):
            print("MASTERCARD")

        else:
            print("INVALID")

    else:
        print("INVALID")


if __name__ == "__main__":
    main()

File: pset1/credit/credit.c
#include<stdio.h>
#include<cs50.h>
#include<math.h>
#include<string.h>
int main(void)
{

    long long card = 0;
    //promote to enter number
    do
    {

        card = get_long_long("Number: ");

    }
    while (card <= 0);

    int counter = 0, numbers = 0;
    //copy of card number
    long long temp;
    temp = card;
    //to help when multiplying number
    int help = 0;
    // firs and second number (last and befor last fom the right)
    int blast = 0, last = 0;
    //x to mutiply every other number from the right
    bool x = false;
    //go thought all numbers in card number
    while (temp > 0)
    {
        numbers++ ;
        if (temp < 100 && temp > 9)
        {
            blast = temp % 10;
        }
        else if (temp < 10 && temp >= 0)
        {
            last = temp;
        }
        help = temp % 10;
        temp /= 10;
        if (x)
        {
            help *= 2;
            if (help > 9)
            {
                help -= 9;
            }
            counter += help;

        }

        else
        {
            counter += help;
        }
        x = !x ;

    }
//if result dividable by 10 check other condition
    if (counter % 10 == 0)
    {
        if (numbers == 15 && last == 3 && (blast == 4 || blast == 7))
        {
            printf("AMEX\n");
        }
        else if ((numbers == 16 || numbers == 13) && last == 4)
        {
            printf("VISA\n");
        }
        else if (numbers == 16 && last == 5 && (blast == 1 || blast == 2 || blast == 3 || blast == 4 || blast == 5))
        {
            printf("MASTERCARD\n");
        }

        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }

    //printf("%i",counter);

}